app_legacy/Helpers/database/player-game.php

```diff
@@ -91,60 +91,193 @@ function getGameRankAndScore(int $gameID, string $username): array
+            }
+
+            if ($numRecentAchievements >= 0) {
+                if ($dateEarnedHardcore !== null) {
+                    $unlockedAchievements[] = [
+                        'Achievement' => $achievement,
+                        'When' => $dateEarnedHardcore,
+                        'Hardcore' => 1,
+                        'Game' => $gameData,
+                    ];
+                } elseif ($dateEarned !== null) {
+                    $unlockedAchievements[] = [
+                        'Achievement' => $achievement,
+                        'When' => $dateEarned,
+                        'Hardcore' => 0,
+                        'Game' => $gameData,
+                    ];
+                } else {
+                    $lockedAchievements[] = [
+                        'Achievement' => $achievement,
+                        'Game' => $gameData,
+                    ];
+                }
+            }
+        }
+
+        $awardedData[$gameID] = [
+            'NumPossibleAchievements' => $numAchievements,
+            'PossibleScore' => $possibleScore,
+            'NumAchieved' => $numAchieved,
+            'ScoreAchieved' => $scoreAchieved,
+            'NumAchievedHardcore' => $numAchievedHardcore,
+            'ScoreAchievedHardcore' => $scoreAchievedHardcore,
+        ];
+
+        if ($withGameInfo) {
+            $gameInfo[$gameID] = [
+                'ID' => (int) $gameData['ID'],
+                'Title' => $gameData['Title'],
+                'ConsoleID' => (int) $gameData['ConsoleID'],
+                'ConsoleName' => $gameData['ConsoleName'],
+                'ForumTopicID' => (int) $gameData['ForumTopicID'],
+                'Flags' => (int) $gameData['Flags'],
+                'ImageIcon' => $gameData['ImageIcon'],
+                'ImageTitle' => $gameData['ImageTitle'],
+                'ImageIngame' => $gameData['ImageIngame'],
+                'ImageBoxArt' => $gameData['ImageBoxArt'],
+                'Publisher' => $gameData['Publisher'],
+                'Developer' => $gameData['Developer'],
+                'Genre' => $gameData['Genre'],
+                'Released' => $gameData['Released'],
+                'IsFinal' => (int) $gameData['IsFinal'],
+            ];
+        }
     }
-    $gameIDs = implode(',', $gameIDs);
-
-    // Count num possible achievements
-    $query = "SELECT GameID, COUNT(*) AS AchCount, SUM(ach.Points) AS PointCount FROM Achievements AS ach
-              WHERE ach.Flags = " . AchievementType::OfficialCore . " AND ach.GameID IN ( $gameIDs )
-              GROUP BY ach.GameID
-              HAVING COUNT(*)>0 ";
-
-    $dbResult = legacyDbFetchAll($query);
-    foreach ($dbResult as $data) {
-        $dataOut[$data['GameID']]['NumPossibleAchievements'] = $data['AchCount'];
-        $dataOut[$data['GameID']]['PossibleScore'] = $data['PointCount'];
+    $libraryOut['Awarded'] = $awardedData;
+
+    if ($withGameInfo) {
+        $libraryOut['GameInfo'] = $gameInfo;
```

Comment:
capturing the game info here saves having to query it again in the calling function